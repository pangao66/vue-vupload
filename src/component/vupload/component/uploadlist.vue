<template>
  <ul>
    <!--<li v-for="(item, index) in thunb" v-if="item" class="item item-list">
      <img :src="list[index] ? list[index] : item.img"  @click="onShow(index)" />
      <div class="del" @click="onDels(index)" v-if="onDel" :style="'display:'+(list[index] ? 'block':'none')"></div>
      <div class="loadImg" v-if="!list[index]">
        <div class="loading"></div>
      </div>
    </li>-->
    
    <li v-for="(item, index) in thunb" :class="'item ' + (list[index] ? 'item-list' : 'item-cache')">
      <img :src="list[index] ? list[index] : item.img"  @click="onShow(index)" />
      <div class="del" @click="onDels(index)" v-if="list[index] ? onDel ? true : false : false"></div>
      <div :class="'loadImg' + ($parent.types.failed == index || failure[index]?' failImg':'')" v-if="!list[index]">
        <div :class="$parent.types.failed == index || failure[index]?'failure':'loading'"></div>
      </div>
    </li>
  </ul>
</template>

<script>
  export default {
    props: {
      data: {
        type: Object,
        default() {
          return {}
        }
      },
     
      list: {
        type: Array,
        default() {
          return []
        }
      },
      thunb: {
        type: Array,
        default() {
          return []
        }
      },
      onDel: {
        type: [Object, Function],
        default: null
      }
    },
    data() {
      return {
        failure: []
      }
    },
    created() {
      for (let i = 0; i < this.$parent.types.max; i++) {
        this.failure.push('')
      }
    },
   
    updated() {
      let num = this.$parent.types.failed
      if (num == -1) return
      this.failure[num] = true
    },
    methods: {
      onShow(index) {
        if (!this.list[index]) return
        let firstImg = this.list[index],
            numImg = this.list[0]
        this.$parent.types.hide = false
        this.$parent.types.pos = index
      },
      onDels(index) {
        this.onDel && this.onDel(index)
      }
    }
  }
</script>
