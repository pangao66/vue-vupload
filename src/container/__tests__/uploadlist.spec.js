import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

import uploadlist from '@/component/vupload/component/uploadlist.vue'

describe('uploadlist.vue', () => {
  it('renders img for each item in props.list', () => {
    const items = ['']
    const renderer = createRenderer()
    const wrapper = shallow(uploadlist, {
      list: items 
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('renders img for each item in props.list null', () => {
    const items = null
    const renderer = createRenderer()
    const wrapper = shallow(uploadlist, {
      propsData: {
        list: items
      } 
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('renders img for each item in props.list length > 0', () => {
    const items = ['https://cmspic-10004025.image.myqcloud.com/9ea8d89b-e5f6-4668-a2d8-754d050aa900?imageMogr2/quality/80/format/webp']
    const wrapper = shallow(uploadlist, {
      propsData: {
        list: items,
        thunb: new Array(items.length)
      } 
    })
    expect(wrapper.findAll('li')).toHaveLength(items.length)
  })


  it('renders img for each item in props.list  thumb && onDel', () => {
    const items = ['https://cmspic-10004025.image.myqcloud.com/9ea8d89b-e5f6-4668-a2d8-754d050aa900?imageMogr2/quality/80/format/webp']
    const renderer = createRenderer()
    const wrapper = shallow(uploadlist, {
      propsData: {
        list: items,
        thunb: items,
        onDel: () => {}
      } 
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })


  it('renders img for each item in props.list li click', () => {
    const items = ['https://cmspic-10004025.image.myqcloud.com/9ea8d89b-e5f6-4668-a2d8-754d050aa900?imageMogr2/quality/80/format/webp']
    const wrapper = shallow(uploadlist, {
      propsData: {
        list: items,
        thunb: items,
        onDel: () => {}
      } 
    })
   const li = wrapper.find('li')
   li.trigger('click')
  })
})