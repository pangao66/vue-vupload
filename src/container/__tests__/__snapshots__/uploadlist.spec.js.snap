// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uploadlist.vue renders img for each item in props.list  thumb && onDel 1`] = `
<ul>
    <li class="item item-list">
        <img src="https://cmspic-10004025.image.myqcloud.com/9ea8d89b-e5f6-4668-a2d8-754d050aa900?imageMogr2/quality/80/format/webp">
        <div class="del"></div>
        <!---->
    </li>
</ul>
`;

exports[`uploadlist.vue renders img for each item in props.list 1`] = `<ul></ul>`;

exports[`uploadlist.vue renders img for each item in props.list null 1`] = `<ul></ul>`;
